diff --git i/plugin/difforig.vim w/plugin/difforig.vim
index 3f37cfd..4568e1b 100644
--- i/plugin/difforig.vim
+++ w/plugin/difforig.vim
@@ -31,7 +31,7 @@ if !(exists(':DiffOrig') == 2)
     command -bar -nargs=0 DiffOrig call difforig#main()
 endif
 
-if !hasmapto('<Plug>(DiffOrig)')
+if !hasmapto('<Plug>(DiffOrig)') && mapcheck('<Leader>o', 'n') is# ''
     nmap <silent> <unique> <Leader>o <Plug>(DiffOrig)
 endif
 nnoremap <unique> <Plug>(DiffOrig) :<C-U>call difforig#main()<CR>
