diff --git a/autoload/fugitive.vim b/autoload/fugitive.vim
index c9c7895..d57416c 100644
--- a/autoload/fugitive.vim
+++ b/autoload/fugitive.vim
@@ -4163,7 +4163,6 @@ function! s:StatusCommand(line1, line2, range, count, bang, mods, reg, arg, args
     let mods = s:Mods(a:mods, 'Edge')
     let file = fugitive#Find(':', dir)
     let arg = ' +setl\ foldmarker=<<<<<<<<,>>>>>>>>' .
-          \ (exists('&winfixbuf') ? '\ winfixbuf' : '') .
           \ '\|let\ w:fugitive_status=FugitiveGitDir() ' .
           \ s:fnameescape(file)
     for tabnr in [tabpagenr()] + (mods =~# '\<tab\>' ? range(1, tabpagenr('$')) : [])
@@ -4178,9 +4177,6 @@ function! s:StatusCommand(line1, line2, range, count, bang, mods, reg, arg, args
             exe winnr . 'wincmd w'
           endif
           let w:fugitive_status = dir
-          if exists('&winfixbuf')
-            setlocal winfixbuf
-          endif
           1
           return ''
         endif
@@ -6076,7 +6072,6 @@ endfunction
 
 function! s:UsableWin(nr) abort
   return a:nr && !getwinvar(a:nr, '&previewwindow') && !getwinvar(a:nr, '&winfixwidth') &&
-        \ !getwinvar(a:nr, '&winfixbuf') &&
         \ (empty(getwinvar(a:nr, 'fugitive_status')) || getbufvar(winbufnr(a:nr), 'fugitive_type') !=# 'index') &&
         \ index(['gitrebase', 'gitcommit'], getbufvar(winbufnr(a:nr), '&filetype')) < 0 &&
         \ index(['nofile','help','quickfix', 'terminal'], getbufvar(winbufnr(a:nr), '&buftype')) < 0
@@ -6208,7 +6203,7 @@ function! fugitive#DiffClose() abort
 endfunction
 
 function! s:BlurStatus() abort
-  if (&previewwindow || getwinvar(winnr(), '&winfixbuf') is# 1 || exists('w:fugitive_status')) && get(b:, 'fugitive_type', '') ==# 'index'
+  if (&previewwindow || exists('w:fugitive_status')) && get(b:, 'fugitive_type', '') ==# 'index'
     let winnrs = filter([winnr('#')] + range(1, winnr('$')), 's:UsableWin(v:val)')
     if len(winnrs)
       exe winnrs[0].'wincmd w'
@@ -7102,9 +7097,6 @@ function! s:BlameSubcommand(line1, count, range, bang, mods, options) abort
         normal! zt
         execute current
         setlocal nonumber scrollbind nowrap foldcolumn=0 nofoldenable winfixwidth
-        if exists('&winfixbuf')
-          setlocal winfixbuf
-        endif
         if exists('+relativenumber')
           setlocal norelativenumber
         endif
