diff --git a/lua/fzf-lua/core.lua b/lua/fzf-lua/core.lua
index 654d722..d2a2b04 100644
--- a/lua/fzf-lua/core.lua
+++ b/lua/fzf-lua/core.lua
@@ -691,132 +691,6 @@ M.set_fzf_field_index = function(opts, default_idx, default_expr)
 end
 
 M.set_header = function(opts, hdr_tbl)
-  local function normalize_cwd(cwd)
-    if path.starts_with_separator(cwd) and cwd ~= vim.loop.cwd() then
-      -- since we're always converting cwd to full path
-      -- try to convert it back to relative for display
-      cwd = path.relative(cwd, vim.loop.cwd())
-    end
-    -- make our home dir path look pretty
-    return path.HOME_to_tilde(cwd)
-  end
-
-  if not opts then opts = {} end
-  if opts.cwd_prompt then
-    opts.prompt = normalize_cwd(opts.cwd or vim.loop.cwd())
-    if tonumber(opts.cwd_prompt_shorten_len) and
-        #opts.prompt >= tonumber(opts.cwd_prompt_shorten_len) then
-      opts.prompt = path.shorten(opts.prompt, tonumber(opts.cwd_prompt_shorten_val) or 1)
-    end
-    if not path.ends_with_separator(opts.prompt) then
-      opts.prompt = opts.prompt .. path.separator()
-    end
-  end
-  if opts.no_header or opts.headers == false then
-    return opts
-  end
-  local definitions = {
-    -- key: opt name
-    -- val.hdr_txt_opt: opt header string name
-    -- val.hdr_txt_str: opt header string text
-    cwd = {
-      hdr_txt_opt = "cwd_header_txt",
-      hdr_txt_str = "cwd: ",
-      hdr_txt_col = opts.hls.header_text,
-      val = function()
-        -- do not display header when we're inside our
-        -- cwd unless the caller specifically requested
-        if opts.cwd_header == false or
-            opts.cwd_prompt and opts.cwd_header == nil or
-            opts.cwd_header == nil and (not opts.cwd or opts.cwd == vim.loop.cwd()) then
-          return
-        end
-        return normalize_cwd(opts.cwd or vim.loop.cwd())
-      end
-    },
-    search = {
-      hdr_txt_opt = "grep_header_txt",
-      hdr_txt_str = "Grep string: ",
-      hdr_txt_col = opts.hls.header_text,
-      val = function()
-        return opts.search and #opts.search > 0 and opts.search
-      end,
-    },
-    lsp_query = {
-      hdr_txt_opt = "lsp_query_header_txt",
-      hdr_txt_str = "Query: ",
-      hdr_txt_col = opts.hls.header_text,
-      val = function()
-        return opts.lsp_query and #opts.lsp_query > 0 and opts.lsp_query
-      end,
-    },
-    regex_filter = {
-      hdr_txt_opt = "regex_header_txt",
-      hdr_txt_str = "Regex filter: ",
-      hdr_txt_col = opts.hls.header_text,
-      val = function()
-        return opts.regex_filter and #opts.regex_filter > 0 and opts.regex_filter
-      end,
-    },
-    actions = {
-      hdr_txt_opt = "interactive_header_txt",
-      hdr_txt_str = "",
-      val = function()
-        if opts.no_header_i then return end
-        local defs = ACTION_DEFINITIONS
-        local ret = {}
-        for k, v in pairs(opts.actions) do
-          local action = type(v) == "function" and v or type(v) == "table" and (v.fn or v[1])
-          if type(action) == "function" and defs[action] then
-            local def = defs[action]
-            local to = (opts.fn_reload or opts._hdr_to) and def.fn_reload or def[1]
-            table.insert(ret, def.pos or #ret + 1,
-              string.format("<%s> to %s",
-                utils.ansi_from_hl(opts.hls.header_bind, k),
-                utils.ansi_from_hl(opts.hls.header_text, to)))
-          end
-        end
-        -- table.concat fails if the table indexes aren't consecutive
-        return not vim.tbl_isempty(ret) and (function()
-          local t = {}
-          for _, i in pairs(ret) do
-            table.insert(t, i)
-          end
-          t[1] = (opts.header_prefix or ":: ") .. t[1]
-          return table.concat(t, opts.header_separator or "")
-        end)() or nil
-      end,
-    },
-  }
-  -- by default we only display cwd headers
-  -- header string constructed in array order
-  if not opts.headers then
-    opts.headers = hdr_tbl or { "cwd" }
-  end
-  -- override header text with the user's settings
-  for _, h in ipairs(opts.headers) do
-    assert(definitions[h])
-    local hdr_text = opts[definitions[h].hdr_txt_opt]
-    if hdr_text then
-      definitions[h].hdr_txt_str = hdr_text
-    end
-  end
-  -- build the header string
-  local hdr_str
-  for _, h in ipairs(opts.headers) do
-    assert(definitions[h])
-    local def = definitions[h]
-    local txt = def.val()
-    if def and txt then
-      hdr_str = not hdr_str and "" or (hdr_str .. ", ")
-      hdr_str = ("%s%s%s"):format(hdr_str, def.hdr_txt_str,
-        not def.hdr_txt_col and txt or
-        utils.ansi_from_hl(def.hdr_txt_col, txt))
-    end
-  end
-  if hdr_str and #hdr_str > 0 then
-    opts.fzf_opts["--header"] = libuv.shellescape(hdr_str)
-  end
   return opts
 end
 
diff --git a/lua/fzf-lua/providers/nvim.lua b/lua/fzf-lua/providers/nvim.lua
index b7a7861..a816e22 100644
--- a/lua/fzf-lua/providers/nvim.lua
+++ b/lua/fzf-lua/providers/nvim.lua
@@ -93,9 +93,6 @@ end
 M.command_history = function(opts)
   opts = config.normalize_opts(opts, config.globals.command_history)
   if not opts then return end
-  if opts.fzf_opts["--header"] == nil then
-    opts.fzf_opts["--header"] = arg_header("<CR>", "<Ctrl-e>", "execute")
-  end
   history(opts, "cmd")
 end
 
