diff --git a/lua/fzf-lua/profiles/ivy.lua b/lua/fzf-lua/profiles/ivy.lua
index cb4ce6d..16281a7 100644
--- a/lua/fzf-lua/profiles/ivy.lua
+++ b/lua/fzf-lua/profiles/ivy.lua
@@ -40,22 +40,4 @@ local M       = {
   },
 }
 
-local up      = {
-  row = 1,
-  col = 0,
-  width = 1,
-  height = 1,
-  preview = {
-    layout = "vertical",
-    vertical = "up:60%",
-    border = "none",
-  },
-}
-
-M.blines      = { winopts = up, previewer = { toggle_behavior = "extend" } }
-M.lines       = M.blines
-M.grep        = M.blines
-M.grep_curbuf = M.blines
-M.git         = { blame = { winopts = up } }
-
 return M
