diff --git i/lua/fzf-lua/actions.lua w/lua/fzf-lua/actions.lua
index 8919899..af444b2 100644
--- i/lua/fzf-lua/actions.lua
+++ w/lua/fzf-lua/actions.lua
@@ -188,7 +188,6 @@ local sel_to_qf = function(selected, opts, is_loclist)
       items = qf_list,
       title = opts.__resume_data.last_query,
     })
-    vim.cmd(opts.lopen or "lopen")
   else
     -- Set the quickfix title to last query and
     -- append a new list to end of the stack (#635)
@@ -198,7 +197,6 @@ local sel_to_qf = function(selected, opts, is_loclist)
       title = opts.__resume_data.last_query,
       -- nr = nr,
     })
-    vim.cmd(opts.copen or "copen")
   end
 end
 
