diff --git a/lua/fzf-lua/actions.lua b/lua/fzf-lua/actions.lua
index 40c37d4..05bf620 100644
--- a/lua/fzf-lua/actions.lua
+++ b/lua/fzf-lua/actions.lua
@@ -682,6 +682,28 @@ end
 
 M.help = function(selected, opts)
   if #selected == 0 then return end
+  local helpcmd
+  local is_shown = false
+  local current_win_number = 1
+  local last_win_number = vim.fn.winnr("$")
+  while current_win_number <= last_win_number do
+    local buffer = vim.api.nvim_win_get_buf(vim.fn.win_getid(current_win_number))
+    local type = vim.api.nvim_get_option_value("buftype", {})
+    if type == "help" then
+      is_shown = true
+      break
+    end
+    current_win_number = current_win_number + 1
+  end
+  if is_shown then
+    helpcmd = "help "
+  else
+    helpcmd = "enew | setlocal bufhidden=wipe | setlocal buftype=help | keepjumps help "
+  end
+  vim.cmd(helpcmd .. helptags(selected, opts)[1])
+end
+
+M.help_split = function(selected, opts)
   vim.cmd("help " .. helptags(selected, opts)[1])
 end
 
diff --git a/lua/fzf-lua/defaults.lua b/lua/fzf-lua/defaults.lua
index 8e407a8..be47ca5 100644
--- a/lua/fzf-lua/defaults.lua
+++ b/lua/fzf-lua/defaults.lua
@@ -781,7 +781,7 @@ M.defaults.awesome_colorschemes = {
 M.defaults.helptags             = {
   actions   = {
     ["enter"]  = actions.help,
-    ["ctrl-s"] = actions.help,
+    ["ctrl-s"] = actions.help_split,
     ["ctrl-v"] = actions.help_vert,
     ["ctrl-t"] = actions.help_tab,
   },
